plugins/variants/src/VariantFeatureWidget/VariantFeatureWidget.js

```diff
@@ -1,106 +1,79 @@
-  },
-  fieldValue: {
-    display: 'inline-block',
-    fontSize: '0.8em',
-  },
-  header: {
-    padding: theme.spacing(0.5),
-    backgroundColor: '#ddd',
-  },
-  title: {
-    fontSize: '1em',
-  },
-
-  valbox: {
-    border: '1px solid #bbb',
-  },
-}))
-
 function VariantSamples(props) {
-  const classes = useStyles()
+  const [filter, setFilter] = useState({})
   const { feature } = props
-  if (!feature.samples) {
+
+  const preFilteredRows = Object.entries(feature.samples || {})
+  if (!preFilteredRows.length) {
     return null
   }
-  const ret = Object.keys(feature.samples)
-  if (!ret.length) {
-    return null
+  const infoFields = ['sample', ...Object.keys(preFilteredRows[0][1])].map(
+    field => ({
+      field,
+    }),
+  )
+  let error
+  let rows = []
+  const filters = Object.keys(filter)
+
+  // catch some error thrown from regex
+  try {
+    rows = preFilteredRows
+      .map(row => {
+        return { sample: row[0], ...row[1], id: row[0] }
+      })
+      .filter(row =>
+        filters.length
+          ? filters.some(key => String(row[key]).match(filter[key] || ''))
+          : true,
+      )
+  } catch (e) {
+    error = e
   }
-  const infoFields = Object.keys(feature.samples[ret[0]])
 
   return (
     <BaseCard {...props} title="Samples">
-      <div style={{ width: '100%', maxHeight: 600, overflow: 'auto' }}>
-        <Table className={classes.table}>
-          <TableHead>
-            <TableRow>
-              <TableCell>Sample</TableCell>
-              {infoFields.map(f => (
-                <TableCell key={f}>{f}</TableCell>
-              ))}
-            </TableRow>
-          </TableHead>
-          <TableBody>
-            {Object.entries(feature.samples).map(
-              ([key, value]) =>
-                value && (
-                  <TableRow key={key}>
-                    <TableCell component="th" scope="row">
-                      {key}
-                    </TableCell>
-                    {infoFields.map(f => (
-                      <TableCell className={classes.valueCell} key={f}>
-                        {value === null ? '.' : String(value[f])}
-                      </TableCell>
-                    ))}
-                  </TableRow>
-                ),
-            )}
-          </TableBody>
-        </Table>
+      {error ? <Typography color="error">{`${error}`}</Typography> : null}
+
+      {infoFields.map(({ field }) => {
+        return (
+          <TextField
+            key={`filter-${field}`}
+            placeholder={`Filter ${field} (regex)`}
+            value={filter[field] || ''}
+            onChange={event =>
+              setFilter({ ...filter, [field]: event.target.value })
+            }
+          />
+        )
+      })}
+      <div style={{ height: 600, width: '100%' }}>
+        <DataGrid
+          autoHeight
+          rows={rows}
+          columns={infoFields}
+          rowHeight={20}
+          headerHeight={25}
+        />
```

Comment:
Can we set the number of rows shown by default? If so, I think 25 rows by default might be sufficient.

Also, when the number of rows shown is high, it looks like it can overflow the background paper:

![image](https://user-images.githubusercontent.com/25592344/101099189-80a0b080-3581-11eb-85cd-0f3171b785f4.png)

Reply:
samples shouldn't overrun this zone now